src/VersionTool/Program.cs

```diff
@@ -0,0 +1,307 @@
+            var paths = pathOption.Values;
+            if (paths.Count == 0)
+            {
+                paths.Add(Path.GetFullPath("."));
+            }
+
+            var matcher = CreateMatcher(dependencyArgument.Value);
+
+            foreach (var project in EnumerateProjects(paths))
+            {
+                var root = LoadJObject(project.ProjectFilePath);
+                foreach (var dependency in EnumerateDependencies(root))
+                {
+                    var matches = new List<JProperty>();
+
+                    if (matcher(dependency.Name) &&
+                        (!matchingOption.HasValue() ||
+                            matchingOption.Values.Contains(dependency.Value.ToString())))
+                    {
+                        matches.Add(dependency);
+                    }
+
+                    if (matches.Count > 0)
+                    {
+                        Console.WriteLine($"Project: {project.ProjectFilePath}");
+                        foreach (var match in matches)
+                        {
+                            Console.WriteLine($"{match.Name}: {match.Value.ToString()}");
+                        }
+
+                        Console.WriteLine();
+                    }
+                }
+            }
+
+            return 0;
+        }
+
+        private static int OnUpdateDependency(
+            CommandOption pathOption,
+            CommandOption matchingOption,
+            CommandArgument dependencyArgument,
+            CommandArgument versionArgument)
+        {
+            var paths = pathOption.Values;
+            if (paths.Count == 0)
+            {
+                paths.Add(Path.GetFullPath("."));
+            }
+
+            var matcher = CreateMatcher(dependencyArgument.Value);
+
+            foreach (var project in EnumerateProjects(paths))
+            {
+                var updated = false;
+
+                var root = LoadJObject(project.ProjectFilePath);
+                foreach (var dependency in EnumerateDependencies(root))
+                {
+                    if (matcher(dependency.Name) &&
+                        (!matchingOption.HasValue() ||
+                            matchingOption.Values.Contains(dependency.Value.ToString())))
+                    {
+                        dependency.Value.Replace(new JValue(versionArgument.Value));
+                        updated = true;
+                    }
+                }
+
+                if (updated)
+                {
+                    SaveJObject(project.ProjectFilePath, root);
+
+                    Console.WriteLine($"Updated: {project.ProjectFilePath}");
+                }
+            }
+
+            return 0;
+        }
+
+        private static IEnumerable<Project> EnumerateProjects(IEnumerable<string> paths)
+        {
+            return paths.SelectMany(p =>
+            {
+                return
+                    Directory.EnumerateFiles(p, "project.json", SearchOption.AllDirectories)
```

Comment:
Could we limit this to `src`, `test`, `samples` and `tools`? I'm not sure if repos have random directories with things that we shouldn't be touching.